File: jjreactnative/ios/JamJaReactNative/libraries/OldUserStorage/OldUserStorageModule.m
//
//  OldUserStorageModule.m
//  JamJaReactNative
//
//  Created by Sinh Phan on 10/7/18.
//  Copyright © 2018 Facebook. All rights reserved.
//

#import "OldUserStorageModule.h"

#import <UIKit/UIKit.h>
#import <React/RCTLog.h>

@implementation OldUserStorageModule

RCT_EXPORT_MODULE(OldUserStorage);

- (NSDictionary<NSString *, id> *)constantsToExport
{
  @try {
    NSUserDefaults *prefs = [NSUserDefaults standardUserDefaults];
    
    if (prefs == nil) return @{};
    
    NSString *userID = [prefs objectForKey:@"userID"];
    NSString *userDisplayName = [prefs objectForKey:@"displayName"];
    NSString *userAvatar = [prefs objectForKey:@"avatar"];
    NSString *userPhone = [prefs objectForKey:@"phone_number"];
    NSString *accessToken = [prefs objectForKey:@"access_token"];
    NSString *tokenType = [prefs objectForKey:@"token_type"];
    NSString *userEmail = [prefs objectForKey:@"email"];
    
    if (userID == nil ||
        accessToken == nil ||
        tokenType == nil) return @{};
    
    if (userDisplayName == nil) userDisplayName = @"";
    if (userAvatar == nil) userAvatar = @"";
    if (userPhone == nil) userPhone = @"";
    if (userEmail == nil) userEmail = @"";
    
    return @{
             @"userId"          : userID,
             @"userDisplayName" : userDisplayName,
             @"userAvatar"      : userAvatar,
             @"userPhone"       : userPhone,
             @"accessToken"     : accessToken,
             @"tokenType"       : tokenType,
             @"userEmail"       : userEmail,
             };
  }
  @catch(NSException* error) {
    
    return @{};
  }
  
}

@end

File: jjreactnative/android/app/src/main/java/com/jamjavn/jamja/module/manager/OldUserStorageModule.java
package com.jamjavn.jamja.module.manager;

import android.content.Context;
import android.content.SharedPreferences;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;

import java.util.HashMap;
import java.util.Map;

import javax.annotation.Nullable;

/**
 * Created by sinhphan on 10/8/18.
 * Email: pnsinh.hg92@gmail.com
 **/
public class OldUserStorageModule extends ReactContextBaseJavaModule {

    public OldUserStorageModule(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {
        return "OldUserStorage";
    }

    @Override
    public @Nullable
    Map<String, Object> getConstants() {

        HashMap<String, Object> constants = new HashMap<>();

        try {
            if (getCurrentActivity() == null || getCurrentActivity().isFinishing()) return constants;

            SharedPreferences sharedPreferences = getCurrentActivity().getSharedPreferences("UserPreference", Context.MODE_PRIVATE);

            if (sharedPreferences == null) return constants;

            String userId = sharedPreferences.getString("userId", "");
            String userToken = sharedPreferences.getString("userToken", "");

            if (userId.isEmpty() || userToken.isEmpty()) return constants;

            userId = userId.trim();

            String[] result = userToken.split("\\s+");

            if (result.length < 2) return constants;

            String tokenType = result[0].trim();
            String accessToken = result[1].trim();

            constants.put("userId", userId);
            constants.put("accessToken", accessToken);
            constants.put("userAvatar", sharedPreferences.getString("avatar", ""));
            constants.put("userEmail", sharedPreferences.getString("email", ""));
            constants.put("userDisplayName", sharedPreferences.getString("displayName", ""));
            constants.put("userPhone", sharedPreferences.getString("phoneNumber", ""));
            constants.put("tokenType", tokenType);
            return constants;
        } catch (Exception e) {
            e.printStackTrace();
            return constants;
        }
    }
}
